src/TableV8/styled.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2020 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { styled } from '../ThemeProvider';
+
+export const StyledTable = styled('div')`
+  border: ${({ theme }) => `solid 1px ${theme.colors.grey_2}`};
+  flex: auto 1;
+  display: flex;
+  flex-direction: column;
+  align-items: stretch;
+  width: 100%;
+  border-collapse: collapse;
+  overflow: auto;
+  * {
+    box-sizing: border-box;
+  }
+`;
+
+export const StyledTbody = styled('div')`
+  min-width: 200px;
+  flex: 99999 1 auto;
+  display: flex;
+  flex-direction: column;
+  overflow: auto;
+`;
+
+export const StyledTrGroup = styled('div')`
+  flex: 1 0 auto;
+  display: flex;
+  flex-direction: column;
+  align-items: stretch;
+`;
+
+export const StyledTr = styled('div')`
+  flex: 1 0 auto;
+  display: inline-flex;
+  background-color: ${({ theme }) => theme.colors.white};
+  &.-even.-striped {
```

Comment:
I prefer this way of styling, having a "big" encapsulating styled component and then smaller classnames. Wondering if you think we should enforce consistency? Curious to hear your thoughts as I've gone back and forth on this a lot.

A lot of usage of Emotion has the styled component reacting to a prop.
eg. 
``` css
background: ${({ theme, striped }) => striped ? stripedBackground : theme.colors.grey_4};
```
where here we're mixing paradigms (which I'm not wholly against) having logic to apply a `className` in our regular component, and really just using Emotion like css modules. We can certainly take this offline as it's a bigger discussion of using Emotion vs scss modules for projects :)

Reply:
Advocating for consistency is always appreciated! I support @ciaranschutte's comment, while I personally am more inclined towards BEM when creating reusable libraries; over the atomic descriptive pattern (e.g. .red.large.rounded) which can get a bit messy in semantics, imo.

Reply:
BEM requires discipline ;p
.. on a green field I would be more than happy to go with BEM + sass modules.
my emotions on emotion have taken a beating

Reply:
thanks for calling this out. I'm on the fence about our CSS approach at the moment, I'll elaborate offline